lib/ansible/module_utils/facts/hardware/hpux.py

```diff
@@ -55,7 +55,7 @@ def get_cpu_facts(self, collected_facts=None):
         cpu_facts = {}
         collected_facts = collected_facts or {}
 
-        if collected_facts.get('ansible_architecture') == '9000/800':
+        if collected_facts.get('ansible_architecture') in ['9000/800', '9000/785']:
```

Comment:
Do we know what the other possible values here are?

Can we generalize and assume any '^9000.*' ansible_architecture should use ioscan?

Could we invert the if/elif check for ia64 and else use ioscan?

roughly...

``` python
if collected.facts.get('ansible_architecture')  == 'ia64':
      # < ia64 stuff here >
else:
     # < 9000 ioscan stuff here>

```

or check for '9000' in arch?
```  python
if collected_facts.get('ansible_architecture') == 'ia64':
    #  < do ia64 stuff >
elif collected_facts.get('ansible_architecture', '').startswith('9000/'):
  # < do 9000 ioscan stuff >
```

Reply:
I am not sure about this. Adding more people - @nrwahl2 @davx8342 - Could you please provide some advice over here ?